### PR TITLE
Err when fails_with compiler argument is unknown

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by sjackman** [comment]
Do not merge this PR until references to `fails_with :gcc => "4.3"` have been removed from Homebrew/core.

**Comment by sjackman** [comment]
I'll open a PR to remove `fails_with :gcc => "4.3"` from Homebrew/core once this PR has been reviewed.

**Comment by sjackman** [comment]
Need to fix
llvm.rb:155
llvm@3.9.rb:61
llvm@4.rb:60
llvm@5.rb:59
llvm@6.rb:67
node.rb:31
node@6.rb:30
node@8.rb:32
rust.rb:53
swift.rb:25

File: Library/Homebrew/software_spec.rb [hunk 1]
```diff
@@ -212,8 +212,9 @@ def patch(strip = :p1, src = nil, &block)
   end
 
   def fails_with(compiler, &block)
-    odisabled "fails_with :llvm" if compiler == :llvm
-    compiler_failures << CompilerFailure.create(compiler, &block)
+    failure = CompilerFailure.create(compiler, &block)
+    odisabled "fails_with #{compiler.inspect}" unless CompilerSelector::COMPILERS.include? failure.name
```
Comment (MikeMcQuaid) [inline Library/Homebrew/software_spec.rb]: Probably want to go for an `odeprecated` first for non-LLVM. Ideally that code would all live in `compat`.

**Pull request closed without merging**